repository.py

```diff
@@ -0,0 +1,543 @@
+#!/usr/bin/env python
+
+# Taken from https://opendev.org/openstack/sunbeam-charms/src/commit/5b37e0a6919668f23b8c7b148717714889fd4381/repository.py
+
+import logging
+import argparse
+import dataclasses
+import pathlib
+import shutil
+import subprocess
+import os
+import glob
+
+import yaml
+
+ROOT_DIR = pathlib.Path(__file__).parent
+EXTERNAL_LIB_DIR = ROOT_DIR / "external" / "lib"
+BUILD_PATH = ROOT_DIR / ".slurm-build"
+BUILD_FILE = ".slurm-build.yaml"
```

Comment:
Hmm... I'm not sure how I feel about a hidden .build file that will dictate how the charm is built. Typically I like build files to be explicit. What are your thoughts on this?

Also, why `.slurm-build.yaml` instead of just `.build.yaml`? Not sure we need the additional namespacing here on the file if repository.py is a general tool that can be used by charm monorepos.

Reply:
I don't want to conflict with some other tool that also uses `build.yaml` for its config. Agreed with using a non-hidden file though.